clients/rust/build.rs

```diff
@@ -1,9 +1,71 @@
 extern crate protoc_rust;
 
+// This is a hack. protoc_rust depends on "protoc" being installed on
+// on the path. Without it, the docs.rs documentation build will fail
+// 
+// As far as I can tell there is no way to know if "cargo doc"
+// was called vs "cargo build". 
+//
+// The profiles sections of Cargo.toml are supposed to help address this
+// but they don't seem to take effect at this stage of the build. 
+// profiles sections: https://doc.rust-lang.org/cargo/reference/manifest.html#the-profile-sections
+//
+// 
+// This workaround allow us to catch a panic from "protoc_rust" if two conditions are met:
```

Comment:
Why do we need to do this?

Reply:
I played around with all the env vars and could not find any that would indicate whether the build was a doc build or a regular build. This is hack, but I spent a lot of time figuring out what didn't work before deciding to go with this.